src/mdns/mdns_mdnssd.cpp

```diff
@@ -1240,6 +1240,21 @@ void PublisherMDnsSd::ServiceSubscription::ProcessAll(const MainloopContext
     }
 }
 
+std::shared_ptr<PublisherMDnsSd::ServiceInstanceResolution>
+PublisherMDnsSd::ServiceSubscription::ExtractResolution(const ServiceInstanceResolution * aResolution)
+{
+    for (auto it = mResolvingInstances.begin(); it != mResolvingInstances.end(); ++it)
+    {
+        if (it->get() == aResolution)
+        {
+            auto ptr = std::move(*it);
+            mResolvingInstances.erase(it);
+            return ptr;
+        }
+    }
```

Comment:
```suggestion
     auto iter = std::find(mResolvingInstances.begin(), mResolvingInstances.end(), aResolution);
     if (iter != mResolvingInstances.end())
     {
            auto ptr = std::move(*iter);
            mResolvingInstances.erase(iter);
            return ptr;
     }
     
     return nullptr;
```
Use std::find? (No strong preference)

Reply:
IIUC, it may not be able to call `==` operator between a raw pointer and a smart pointer.